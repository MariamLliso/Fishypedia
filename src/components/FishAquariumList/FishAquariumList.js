import Fish from "../Fish/Fish";
import styled from "styled-components";

const FishListStyle = styled.ul`
  display: flex;
  flex-direction: column;
  gap: 70px;
  margin-top: 150px;
  margin-bottom: 30px;
`;

const FishAquariumList = ({ fishes }) => {
  return (
    <>
      <FishListStyle>
        {fishes.slice(0, 9).map((fish) => (
          <li className="fish" key={fish.id}>
            <Fish fish={fish} />
          </li>
        ))}
      </FishListStyle>
    </>
  );
};

export default FishAquariumList;
