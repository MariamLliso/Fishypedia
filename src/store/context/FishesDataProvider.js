import { useState, useReducer } from "react";
import fishesReducer from "../reducers/fishesReducer";
import localFishesReducer from "../reducers/localFishesReducer";
import FishesDataContext from "./FishesDataContext";

const FishesDataProvider = ({ children }) => {
  const [fishes, dispatch] = useReducer(fishesReducer, []);
  const [localFishes, localDispatch] = useReducer(localFishesReducer, []);
  const initialPage = "";
  const [page, setPage] = useState(initialPage);

  return (
    <FishesDataContext.Provider
      value={{ fishes, dispatch, localFishes, localDispatch, page, setPage }}
    >
      {children}
    </FishesDataContext.Provider>
  );
};

export default FishesDataProvider;
