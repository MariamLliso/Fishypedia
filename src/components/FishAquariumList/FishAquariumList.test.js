import FishesDataProvider from "../../store/context/FishesDataProvider";
import FishAquariumList from "./FishAquariumList";
import TestRenderer from "react-test-renderer";

describe("Given a FishAquariumList component", () => {
  describe("When its rendered with 2 fishes", () => {
    test("Then it should always match this snapshot", () => {
      const fishes = [
        {
          id: 1,
          "file-name": "bitterling",
          name: {
            "name-USen": "bitterling",
            "name-EUen": "bitterling",
            "name-EUde": "Bitterling",
            "name-EUes": "amarguillo",
            "name-USes": "amarguillo",
            "name-EUfr": "bouvière",
            "name-USfr": "bouvière",
            "name-EUit": "rodeo",
            "name-EUnl": "bittervoorn",
            "name-CNzh": "红目鲫",
            "name-TWzh": "紅目鯽",
            "name-JPja": "タナゴ",
            "name-KRko": "납줄개",
            "name-EUru": "горчак",
          },
          availability: {
            "month-northern": "11-3",
            "month-southern": "5-9",
            time: "",
            isAllDay: true,
            isAllYear: false,
            location: "River",
            rarity: "Common",
            "month-array-northern": [11, 12, 1, 2, 3],
            "month-array-southern": [5, 6, 7, 8, 9],
            "time-array": [
              0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18,
              19, 20, 21, 22, 23,
            ],
          },
          shadow: "Smallest (1)",
          price: 900,
          "price-cj": 1350,
          "catch-phrase":
            "I caught a bitterling! It's mad at me, but only a little.",
          "museum-phrase":
            "Bitterlings hide their eggs inside large bivalves—like clams—where the young can stay safe until grown. The bitterling isn't being sneaky. No, their young help keep the bivalve healthy by eating invading parasites! It's a wonderful bit of evolutionary deal making, don't you think? Each one keeping the other safe... Though eating parasites does not sound like a happy childhood... Is that why the fish is so bitter?",
          image_uri: "https://acnhapi.com/v1/images/fish/1",
          icon_uri: "https://acnhapi.com/v1/icons/fish/1",
        },
        {
          id: 2,
          "file-name": "pale_chub",
          name: {
            "name-USen": "pale chub",
            "name-EUen": "pale chub",
            "name-EUde": "Döbel",
            "name-EUes": "cacho",
            "name-USes": "cacho",
            "name-EUfr": "chevaine",
            "name-USfr": "chevaine",
            "name-EUit": "zacco",
            "name-EUnl": "Japanse drakenvis",
            "name-CNzh": "溪哥",
            "name-TWzh": "溪哥",
            "name-JPja": "オイカワ",
            "name-KRko": "피라미",
            "name-EUru": "светлый закко",
          },
          availability: {
            "month-northern": "",
            "month-southern": "",
            time: "9am - 4pm",
            isAllDay: false,
            isAllYear: true,
            location: "River",
            rarity: "Common",
            "month-array-northern": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
            "month-array-southern": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
            "time-array": [9, 10, 11, 12, 13, 14, 15],
          },
          shadow: "Smallest (1)",
          price: 200,
          "price-cj": 300,
          "catch-phrase":
            "I caught a pale chub! That name seems a bit judgy...",
          "museum-phrase":
            "The pale chub is a river fish with simple back-and-silver coloring. Interestingly, the males' coloring becomes most vibrant when he's trying to attract a mate! Though normally plain, these fellows really know how to look stylish when they want to. Perhaps I could learn a thing or two from the pale chub...",
          image_uri: "https://acnhapi.com/v1/images/fish/2",
          icon_uri: "https://acnhapi.com/v1/icons/fish/2",
        },
      ];

      const renderedFishComponent = TestRenderer.create(
        <FishesDataProvider>
          <FishAquariumList fishes={fishes} />
        </FishesDataProvider>
      );

      expect(renderedFishComponent).toMatchSnapshot();
    });
  });
});
