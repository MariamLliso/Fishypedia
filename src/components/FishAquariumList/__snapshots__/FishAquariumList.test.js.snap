// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a FishAquariumList component When its rendered with 2 fishes Then it should always match this snapshot 1`] = `
<ul
  className="sc-dkzDqf jxeCkQ"
>
  <li
    className="fish"
  >
    <div
      className="sc-bczRLJ cmDTID"
    >
      <ul
        className="fish-container"
      >
        <li
          className="fish-container__header"
        >
          <h3
            className="fish-container__name"
          >
            Bitterling
          </h3>
          <div
            className="fish-icon__container"
          >
            <img
              alt="bitterling image"
              className="fish-icon__image"
              height="80px"
              src="https://acnhapi.com/v1/icons/fish/1"
              width="auto"
            />
          </div>
        </li>
        <li>
          <div
            className="fish-container__monthDisplay"
          >
            <i
              className="fas fa-calendar-alt"
            />
            <span>
              11-3
            </span>
          </div>
          <div
            className="fish-container__hourDisplay"
          >
            <i
              className="fas fa-clock"
            />
            <span>
              All day
            </span>
          </div>
        </li>
        <li>
          <div
            className="fish-container__locationDisplay"
          >
            <span>
              River
            </span>
          </div>
          <div
            className="fish-container__rarityDisplay"
          >
            <span>
              Common
            </span>
          </div>
        </li>
        <li
          className="fish-container__button"
        />
      </ul>
    </div>
  </li>
  <li
    className="fish"
  >
    <div
      className="sc-bczRLJ cmDTID"
    >
      <ul
        className="fish-container"
      >
        <li
          className="fish-container__header"
        >
          <h3
            className="fish-container__name"
          >
            Pale chub
          </h3>
          <div
            className="fish-icon__container"
          >
            <img
              alt="pale chub image"
              className="fish-icon__image"
              height="80px"
              src="https://acnhapi.com/v1/icons/fish/2"
              width="auto"
            />
          </div>
        </li>
        <li>
          <div
            className="fish-container__monthDisplay"
          >
            <i
              className="fas fa-calendar-alt"
            />
            <span>
              All year
            </span>
          </div>
          <div
            className="fish-container__hourDisplay"
          >
            <i
              className="fas fa-clock"
            />
            <span>
              9am - 4pm
            </span>
          </div>
        </li>
        <li>
          <div
            className="fish-container__locationDisplay"
          >
            <span>
              River
            </span>
          </div>
          <div
            className="fish-container__rarityDisplay"
          >
            <span>
              Common
            </span>
          </div>
        </li>
        <li
          className="fish-container__button"
        />
      </ul>
    </div>
  </li>
</ul>
`;
